src/Spawner.cpp

```diff
@@ -272,6 +283,13 @@ Spawner::ProcessHandle& Spawner::spawnDeployment(Deployment* deployment, bool re
         logDir = Bundle::getInstance().getLogDirectory();
     }
 
+    // rename the logger of default deployments
+    // this guarantees that every task has it's own logger
+    if(deployment->getName().find("orogen_default_") == 0)
+    {
+        deployment->renameTask(deployment->getLoggerName(), deployment->getName() + "_Logger");
+    }
+
     ProcessHandle *handle = new ProcessHandle(deployment, redirectOutput, logDir);
     
     handles.push_back(handle);
```

Comment:
I think handles never get removed from the `handles`-variable. Thus  there's no real cleanup happening. This problem apparently was present also before the MR, but with the issue described at `alive` this might lead to unexpected behavior...

e.g. A single deployment was killed, afterwards `killAll` is called. In this case `alive` will be called on the already-terminated process again.

Reply:
This seems to be the case, I need to look into this a bit further. But I think this issue existed before already.